lib/trento/application/integration/checks/checks.ex

```diff
@@ -7,6 +12,52 @@ defmodule Trento.Integration.Checks do
   def request_execution(execution_id, cluster_id, hosts, selected_checks),
     do: adapter().request_execution(execution_id, cluster_id, hosts, selected_checks)
 
+  @spec get_catalog ::
+          {:ok, FlatCatalog.t()} | {:error, any}
+  def get_catalog do
+    case adapter().get_catalog() do
+      {:ok, catalog} ->
+        FlatCatalog.new(%{checks: catalog})
```

Comment:
we should return the `FlatCatalog.t()` directly from the adapter, this way we enforce a contract with the adapter and we check that the contract was respected before consuming it.

Reply:
Perfect!